### DISCUSSION TITLE
How does the autosave work when different users working on the article (not on the same time)?

### DISCUSSION THREAD

**Discussion by kskonovalov** [elementor/elementor#20252] [Developers Community]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

The issue is, when few persons could edit the page (not the same time of course), some of the changes **could lost**, and it's really frustrating.

**question 1:** Sometimes when I open the page, I see the 'Update' button is green active, even if the last time I saved the page the button was grey inactive. And I don't know, why's that? Does the page has the unsaved changes? Or it loaded just mine 'old' changes, which are already overwritten by somebody else?

My idea is, when the Elementor saves the revision (unsaved state), it may load it and overwrite the last changes which was made and saved on the page.

**question 2:** What will be if, for example, person 1 make some changes but wouldn't save the page. Then person 2 will do the changes and save the page. And if person 1 open the page again -- will it load the actual changes (from person 2), or his (person 1) unsaved changes?

The more or less 'stable' way for me is, when I know somebody was working on the page, and if the 'Update' button is active, maybe it's worth to restore the latest revision from the revisions, but it's not the best way.

(The reason I'm posting it here, and not using the support system, is to hear the community opinion as well, and not only the support reply, because the issue is floating: sometimes it saves the data and sometimes it's not)

Any help/reply/ideas greatly appreciated

### Describe the solution you'd like

More straight and clear flow about saving the pages

**Comment by stasonua0** [comment]
I have observed similar problems, which is why they refused to work as a team on one project, since someone necessarily overwritten the changes. 
At the same time, we created our own account for everyone and paid attention to the warning "Someone is already editing this page."

**Comment by nicholaszein** [comment]
Hey there, @kskonovalov!

This is more of a support question than something that should be discussed inside an issue. You could also have asked this on the [Discussions](https://github.com/elementor/elementor/discussions) forum, so other users could give you this answer.

But let me try to explain:

Elementor uses the default behavior used by WordPress. There is nothing different going on. 

- If a user is editing a document, another user can't edit it themselves. 
- If a user does not save his edits, and then a second user "takes over", edit, and save the document, his changes will override anything else made by the first user. 
- If the first user then "takes over" again and saves his changes without refreshing the page, then his changes will override the second user's changes, and mismatched revisions will be created by WordPress.
- If two people use the same username to make changes, or if you apply different changes to the same document but in different browser tabs/windows, the most recent change will always override the other changes.
- There is no engine or feature that synchronizes changes or allows multiple users to work on the same document simultaneously.

This is how WordPress always worked and Elementor has no intent to change this behavior for the time being, since this could affect compatibility for other plugins that rely on WordPress' default behaviors to work properly.

You can learn more about the Post Locking engine [here](https://codex.wordpress.org/Post_Locking).

As a rule of thumb: 
1. Avoid having multiple users edit the same post back and forward.
2. Do not share the same login with multiple people. There is a complete user management interface just for that. Each user should have their own login.
3. If you need to edit a document that is currently being edited by someone else, ask them to save and close their editing session so you can start editing yourself. Taking over without letting people know is a recipe for disaster.

Until WordPress changes its editing and Post Locking engine, Elementor won't modify the default behavior.

I hope this has clarified your questions.

I will migrate this issue over to  [Discussions](https://github.com/elementor/elementor/discussions).

Kind regards,

Nick
